dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/ThreadCreation.java

```diff
@@ -334,25 +334,60 @@ private ThreadData createLLVMThreadFromFunction(Function function, int tid, Thre
         return new ThreadData(thread, null);
     }
 
-    private void replaceGlobalsByThreadLocals(Memory memory, Thread thread) {
+
+    private void replaceThreadLocalsWithStackalloc(Memory memory, Thread thread) {
+        final TypeFactory types = TypeFactory.getInstance();
+        final ExpressionFactory exprs = ExpressionFactory.getInstance();
+
+        // Translate thread-local memory object to local stack allocation
+        Map<MemoryObject, Register> toLocalRegister = new HashMap<>();
+        for (MemoryObject memoryObject : memory.getObjects()) {
+            if (!memoryObject.isThreadLocal()) {
+                continue;
+            }
+            Preconditions.checkState(memoryObject.hasKnownSize());
```

Comment:
Do all thread local objects have known size? Can't we have a thread local array of unknown size?

Reply:
Thread-locals are declared like global variables. Global variables always have known size (how could they not?).
Variable-sized arrays can only be mimicked by thread-local pointers (known size) to variable-sized arrays.